test/unit/Validation/Constraint/PermittedForTest.php

```diff
@@ -74,4 +74,28 @@ public function assertShouldNotRaiseExceptionWhenAudienceMatches()
         $constraint->assert($token);
         $this->addToAssertionCount(1);
     }
+
+    /**
+     * @test
+     *
+     * @covers ::__construct
+     * @covers ::assert
+     */
+    public function assertShouldAudienceIsArrayWithSingleElement()
+    {
+        // works
+        $aud        = uniqid('aud_', true);
+        $token      = $this->buildToken([RegisteredClaims::AUDIENCE => $aud]);
+        $constraint = new PermittedFor($aud);
+
+        $constraint->assert($token);
+        $this->addToAssertionCount(1);
+
+        // doesnt work
+        $token      = $this->buildToken([0 => [RegisteredClaims::AUDIENCE => $aud]]);
```

Comment:
This should be like this right?
```suggestion
        $token      = $this->buildToken([RegisteredClaims::AUDIENCE => [$aud]]);
```

Reply:
I tried to recreate how our jwt is parsed in the test. As you can see in the screenshots its parsed like [0 => ] [] https://imgur.com/a/Co0l1Ss

Reply:
That's what my suggestion does :)

`[0 => [RegisteredClaims::AUDIENCE => $aud]]` would create a DataSet with the claim `0` which isn't exactly what you want 👍